File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        
        // 리스트에 제품을 수량만큼 추가
        List<String> total=new ArrayList<>();
        
        for(int i=0;i<want.length;i++){
            for(int j=0;j<number[i];j++){
                total.add(want[i]);
            }
        }
        
        for(int i=0; i<discount.length-9; i++){
            // Arrays.copyOfRange(배열, 시작, 종료): 배열 복사
            List<String> sub = new ArrayList<>(Arrays.asList(Arrays.copyOfRange(discount, i, i+10)));
            
            for(int j=0; j<total.size(); j++){
                // 원하는 제품이 할인 목록에 있는지 체크
                sub.remove(total.get(j));
            }
                            
            // 모든 제품이 할인 목록에 있으면 answer 증가
            if(sub.isEmpty()){
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    
    item_dict = dict(zip(want, number))      # {제품:수량}
    count = 0
    
    for i in range(len(discount)-9):
        count = 0
        sales_item = discount[i:i+10]
        
        for k, v in item_dict.items():
            if sales_item.count(k) == v:
                count += 1
            if count == len(want):
                answer += 1
    
    return answer